<a href="https://colab.research.google.com/github/oscardominguez-ds/Explaining-Datasets/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
#mounting drive
from google.colab import drive

drive.mount('/content/drive')


Mounted at /content/drive


##Load data - Adult income dataset




In [4]:
## Load the data
import pandas as pd
df = pd.read_csv('/content/adult.csv')
df.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


##1. Source of data
  -https://www.kaggle.com/datasets/wenruliu/adult-income-dataset?resource=download

##2. Brief description of data
  -An individual’s annual income results from various factors. Intuitively, it is influenced by the individual’s education level, age, gender, occupation, and etc.

##3. What is the target?
  -Income
##4. What does one row represent? (A person?  A business?  An event? A product?)
  -Each row represents an entry from a person with their background information(columns)
##5. Is this a classification or regression problem?
-This is a Regression problem becase we are answering 'how much' someone makes.
##6. How many features does the data have?
15
##7. How many rows are in the dataset?
48842
##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
-Cleaning: checking and removing duplicate entries, and type conversion if say an entry that should be numerical is an object. I don't see challenges in exploring or modeling.


##Load data - Heart Failure Prediction Dataset

In [10]:
## Load the data
import pandas as pd
df = pd.read_csv('/content/heart.csv')
df.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


##1. Source of data
https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction

##2. Brief description of data
-This dataset describes how likely someone is of cardiovascular disease for early detection and management.  

##3. What is the target?
  -HeartDisease
##4. What does one row represent? (A person?  A business?  An event? A product?)
  -Each row represents 1 person 

##5. Is this a classification or regression problem?
-This is classification problem because we want to find out if someone is at at risk of cardiovascular disease.
##6. How many features does the data have?
12
##7. How many rows are in the dataset?
918
##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
-cleaning:  checking and removing duplicate entries and  type conversion if say an entry that should be numerical is an object. I don't see  challenges in exploring or modeling.